db/migrate/20221216171738_value_overrides_to_non_canonical_profiles.rb

```diff
@@ -0,0 +1,13 @@
+class ValueOverridesToNonCanonicalProfiles < ActiveRecord::Migration[7.0]
+  def up
+    Profile.canonical.find_each do |canonical_profile|
+      canonical_profile.child_profiles.update_all(
+        value_overrides: canonical_profile.value_overrides
+      )
+    end
```

Comment:
You are iterating through canonical profiles and then querying their child profiles one-by-one. Try to catch this from the other end, request all non-canonical profiles, join them with their parent profile and use [`update_all`](https://stackoverflow.com/questions/5224881/update-one-column-to-value-of-another-in-rails-migration) to ask PG to do the heavy lifting for you. 

Also if you're using DB models in your migrations, you have to make sure those models are defined in the migration context itself. Right now it would work, but what if in 2 years we change the `Profile` model and someone wants to set up the DB by running migrations on an empty DB (or ephemeral).